tests/integration/StorageWriter/IPFS.test.ts

```diff
@@ -0,0 +1,76 @@
+import { describe } from 'riteway'
+import fetch from 'node-fetch'
+
+import { IPFS } from '../../../src/StorageWriter/IPFS'
+import { allAsciiCharactersClaim, nonAsciiCharactersClaim, longWithNonAsciiCharactersClaim } from './claimData'
+
+const IPFS_URL = process.env.IPFS_URL || 'http://localhost:5001'
+
+const createIPFS = ({
+  ipfsUrl = IPFS_URL
+} = {}) => {
+  return new IPFS({
+    ipfsUrl
+  })
+}
+
+const fetchFile = async (hash: string): Promise<string> => {
+  const response = await fetch(`${IPFS_URL}/api/v0/cat?arg=${hash}`)
+  return response.text()
+}
+
+describe('IPFS.addText', async should => {
+  const { assert } = should('match the claim read from ipfs')
+
+  {
+    const ipfs = createIPFS()
+    const claim = allAsciiCharactersClaim
+    let claimFromIPFS
+    
+    try {
```

Comment:
Not sure if these try {} finally {}'s are needed anymore. I added them because when ipfs was failing it was not returning json and causing node to exit before running the remaining text. Let me know your thoughts, I am good removing them or keeping them.